src/repl/config.rs

```diff
@@ -0,0 +1,87 @@
+use config::{Config, FileFormat, Map};
+use directories::ProjectDirs;
+use std::fs::OpenOptions;
+use std::io::Write;
+use std::path::{Path, PathBuf};
+
+pub(crate) static ION_SYNTAX: &str = include_str!(concat!(
+    env!("CARGO_MANIFEST_DIR"),
+    "/resources/ion.sublime-syntax"
+));
+pub(crate) static PARTIQL_SYNTAX: &str = include_str!(concat!(
+    env!("CARGO_MANIFEST_DIR"),
+    "/resources/partiql.sublime-syntax"
+));
+pub(crate) static DEFAULT_CONFIG: &str = include_str!(concat!(
+    env!("CARGO_MANIFEST_DIR"),
+    "/resources/partiql-cli.toml"
+));
+
+pub(crate) struct ReplConfig {
+    pub config: Config,
+    pub config_dir: PathBuf,
+    pub cache_dir: PathBuf,
+    pub history_path: PathBuf,
+}
+
+pub(crate) fn repl_config() -> ReplConfig {
+    let dirs = ProjectDirs::from("org", "partiql", "partiql-cli").expect("project directories");
+    let config_dir = dirs.config_dir();
+    let cache_dir = dirs.cache_dir();
+    std::fs::create_dir_all(&config_dir).expect("create config dir");
+
+    let mut conf = config_dir.to_path_buf();
+    conf.push("partiql-cli.toml");
+
+    // If the config file does not exist, create it and write the default config into it.
+    //    create_new returns `Ok` if it creates and `Err` if the file already exists
+    if let Ok(mut f) = OpenOptions::new().write(true).create_new(true).open(&conf) {
+        f.write_all(DEFAULT_CONFIG.as_bytes());
+    }
+
+    let config = Config::builder()
+        .set_default("repl", Map::from([("theme".to_string(), infer_theme())]))
+        .expect("default values")
+        .add_source(config::File::from_str(DEFAULT_CONFIG, FileFormat::Toml))
+        .add_source(config::File::new(conf.to_str().unwrap(), FileFormat::Toml))
+        .build()
+        .expect("configuration files");
+
+    // History file used to be written to `~/partiql_cli.history`
+    // If the new history file doesn't exist, but legacy does, copy legacy to new location
+    // TODO remove all the legacy history path stuff after several versions, say v0.6
```

Comment:
Can probably document in an GitHub issue that we intend to remove the old history path logic